src/bun.js/bindings/sqlite/JSSQLStatement.cpp

```diff
@@ -1764,6 +1764,110 @@ JSC_DEFINE_HOST_FUNCTION(jsSQLStatementFcntlFunction, (JSC::JSGlobalObject * lex
 
     return JSValue::encode(jsNumber(statusCode));
 }
+typedef struct {
+    Strong<JSC::JSGlobalObject> globalObject;
+    Strong<JSC::JSFunction> callback;
+} UpdateHookData;
+
+UpdateHookData * data = nullptr;
+
+void on_update_hook(void * dataPtr, int type, const char * dbName, const char * tableName, sqlite_int64 rowId) {
+    UpdateHookData* data = (UpdateHookData *)dataPtr;
+
+    auto globalObject = data->globalObject.get();
```

Comment:
This will need an exception check

```suggestion
    auto globalObject = data->globalObject.get();
    auto &vm = globalObject->vm();
    auto scope = DECLARE_THROW_SCOPE(vm);
    RETURN_IF_EXCEPTION(scope, void());
```